util/check_tool_requirements.py

```diff
@@ -6,50 +6,15 @@
 from distutils.version import StrictVersion
 import logging as log
 import os
-import re
 import subprocess
 import sys
 
 # Display INFO log messages and up.
 log.basicConfig(level=log.INFO, format="%(levelname)s: %(message)s")
 
-
-def get_tool_requirements_path():
-    '''Return the path to tool_requirements.py, at the top of the Ibex repo'''
-    # top_src_dir is the top of the repository
-    top_src_dir = os.path.normpath(os.path.join(os.path.dirname(__file__),
-                                                '..'))
```

Comment:
This patch seems to revert commit `db6f8f0`. Is that intentional?

Reply:
No, that's something I missed. These changes need to go into OT first now, I'll open a PR there.

Reply:
I'm confused. Is `util/check_tool_requirements.py` vendored from the OT repo?

Reply:
You have every reason to be confused.

Before, `util/check_tool_requirements.py` was a diverged copy from OT with a different core name. 

We now need to make sure we have the same core name for this functionality between OT and Ibex, as there are dependencies on it in other core files. With the same core name, I also wanted to make sure that we actually have the same functionality.

So what this PR does is align the core file and the contents of the script with what we have in OT, but we cannot easily vendor it in until https://github.com/lowRISC/opentitan/issues/2309 is fixed.

Reply:
In which case, isn't it better to keep the existing functionality and add it to the OT version once that's possible? At the moment, this patch is deleting code when it's not technically needed, I think.

Reply:
We need to have the core name of the OT version, since that's used in dependencies. And I don't want to use the same core name for cores with different implementations between OT and Ibex -- doing stuff like that got us into this mess in the first place.

I'll update this PR to keep the original functionality once https://github.com/lowRISC/opentitan/pull/2319 goes in, so we don't loose anything.

Reply:
Fixed now.